homeassistant/components/homekit/util.py

```diff
@@ -34,6 +34,11 @@ def validate_entity_config(values):
             code = config.get(ATTR_CODE)
             params[ATTR_CODE] = cv.string(code) if code else None
 
+        if domain == 'media_player':
+            mode = config.get(CONF_MODE)
+            if mode:
+                params[CONF_MODE] = cv.ensure_list(mode)
```

Comment:
You can improve the config validation by checking if only allowed modes are specified:
```py
params[CONF_MODE] = cv.ensure_list(mode)
for mode in params[CONF_MODE]:
    if not mode in ALL_MODES:
        raise vol.Invalid('Invalid mode')
```
`ALL_MODES` being a tuple and the error message being a bit better.